src/resource/ensemble.rs

```diff
@@ -1,9 +1,17 @@
 //! An ensemble of multiple predictive models.
 
+use std::collections::HashMap;
+
 use super::Resource;
 use super::id::*;
 use super::status::*;
 
+/// List of field codes mapped to input fields
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct EnsembleField {
+  pub name: String,
+}
```

Comment:
Also, please configure your editor for 4-space indents, or just run `cargo fmt` to apply standard Rust style rules.